src/StudentWork/Example.hs

```diff
@@ -1 +1,13 @@
 module StudentWork.Example where
+
+data Exp a where
+    EConst :: Int -> Exp a
+    EVar :: a -> Exp a
+
+mapExp :: (a -> b) -> Exp a -> Exp b
+mapExp _ (EConst x) = EConst x
+mapExp f (EVar x)   = EVar (f x) 
+
+foldrExp :: (Exp a -> Exp b -> Exp b) -> Exp b -> [Exp a] -> Exp b
```

Comment:
This is a fold for lists of expressions, but the point is to make a fold function for the expression type itself.